packages/ui/core-components/src/lib/unsorted/viz/references/ReferenceArea/ReferenceArea.svelte

```diff
@@ -3,36 +3,173 @@
+	export let borderType = 'dashed';
+
+	/** @type {import('../types.js').PresetColor | undefined}*/
+	export let borderColor = undefined;
+
+	/** @type {number | string | undefined} */
+	export let borderWidth = undefined;
+
+	/** @type {import('../types.js').PresetColor | undefined} */
+	export let labelColor = undefined;
+
+	/**
+	 * @type {number | string}
+	 * @default 1
+	 */
+	export let labelPadding = 1;
+
+	/** @type {import('./reference-area.js').LabelPosition | undefined} */
+	export let labelPosition = undefined;
+
+	/** @type {string | undefined} */
+	export let labelBackgroundColor = undefined;
+
+	/** @type {number | string | undefined} */
+	export let labelBorderWidth = undefined;
+
+	/** @type {number | string | undefined} */
+	export let labelBorderRadius = undefined;
+
+	/** @type {string | undefined} */
+	export let labelBorderColor = undefined;
+
+	/** @type {'solid' | 'dotted' | 'dashed' | undefined} */
+	export let labelBorderType = undefined;
+
+	/** @type {number | string | undefined} */
+	export let fontSize = undefined;
+
+	/** @type {'left' | 'center' | 'right' | undefined} */
+	export let align = undefined;
+
+	/** @type {boolean | string | undefined} */
+	export let bold = undefined;
+
+	/** @type {boolean | string | undefined} */
+	export let italic = undefined;
+
+	/**
+	 * @type {boolean | string}
+	 * @default false
+	 */
+	export let preserveWhitespace = false;
 
 	let chartType = 'Reference Area';
 
-	// Remove any undefined props (e.g. w/o defaults) to prevent them from being passed
-	$: spreadProps = {
-		...Object.fromEntries(Object.entries($$props).filter(([, v]) => v !== undefined))
-	};
+	// Accept label from slot
+	/** @type {HTMLElement | undefined} */
+	let slotElement = undefined;
+	$: if (slotElement?.textContent) {
+		if (preserveWhitespace) {
+			label = slotElement.textContent;
+		} else {
+			label = slotElement.textContent
+				.split('\n')
+				.map((line) => line.trim())
+				.join('\n');
+		}
+	}
+
+	const initialHash = Query.isQuery(data) ? data.hash : undefined;
+	$: isInitial = Query.isQuery(data) && data.hash === initialHash;
+
+	const props = getPropContext();
+	const config = getConfigContext();
+	const store = createReferenceAreaStore(props, config);
+
+	// React to the props store to make sure the ReferencePoint is added after the chart is fully rendered
+	$: $props,
+		($store = {
+			xMin,
+			xMax,
+			yMin,
+			yMax,
+			data,
+			label,
+			color,
+			areaColor,
+			opacity: toNumber(opacity),
+			border: toBoolean(border),
+			borderType,
+			borderColor,
+			borderWidth: toNumber(borderWidth),
+			labelColor,
+			labelPadding: toNumber(labelPadding),
+			labelPosition,
+			labelBackgroundColor,
+			labelBorderColor,
+			labelBorderWidth: toNumber(labelBorderWidth),
+			labelBorderRadius: toNumber(labelBorderRadius),
+			labelBorderType,
+			fontSize: toNumber(fontSize),
+			align,
+			bold: toBoolean(bold),
+			italic: toBoolean(italic)
+		});
```

Comment:
Similarly, we haven't overloaded the `set` function in the other stores, but have had more specific functions that describe what the behavior is